src/io/flutter/view/WidgetPerfTable.java

```diff
@@ -0,0 +1,518 @@
+      lastActive = i;
+    }
+    if (lastActive >= 0) {
+      final int[] active = new int[lastActive + 1];
+      for (int i = 0; i <= lastActive; i++) {
+        active[i] = i;
+      }
+      model.nodesChanged(root, active);
+    }
+  }
+
+  private void selectionListener(TreeSelectionEvent event) {
+    if (pendingSelectionChangedActions > 0) {
+      return;
+    }
+    if (!event.isAddedPath()) {
+      // We only care about selection events not deselection events.
+      return;
+    }
+    pendingSelectionChangedActions++;
+    final DefaultMutableTreeNode selection = (DefaultMutableTreeNode)event.getPath().getLastPathComponent();
+    if (selection != null) {
+      final SlidingWindowStatsSummary stats = (SlidingWindowStatsSummary)selection.getUserObject();
+      if (stats != null) {
+        final Location location = stats.getLocation();
+        final XSourcePosition position = location.getXSourcePosition();
+        if (position != null) {
+          AsyncUtils.invokeLater(() -> {
+            position.createNavigatable(app.getProject()).navigate(false);
+            pendingSelectionChangedActions--;
+          });
+          return;
+        }
+      }
+    }
+    pendingSelectionChangedActions--;
+  }
+
+  @Override
+  public TableCellRenderer getCellRenderer(int row, int column) {
+    final ColumnInfo m = modelColumns[column];
+    return m.getRenderer(null);
+  }
+
+  private ActionGroup createTreePopupActions() {
+    final DefaultActionGroup group = new DefaultActionGroup();
+    final ActionManager actionManager = ActionManager.getInstance();
+    group.add(actionManager.getAction(InspectorActions.JUMP_TO_SOURCE));
+    return group;
+  }
+
+  ListTreeTableModelOnColumns getTreeModel() {
+    return (ListTreeTableModelOnColumns)getTableModel();
+  }
+
+  public void markAppIdle() {
+    idle = true;
+    countColumnInfo.setIdle(true);
+    updateIconUIAnimations();
+  }
+
+  public void showStats(ArrayList<SlidingWindowStatsSummary> entries) {
+    pendingSelectionChangedActions++;
+    idle = false;
+    countColumnInfo.setIdle(false);
+    final ArrayList<SlidingWindowStatsSummary> oldEntries = this.entries;
+    sortByMetric(entries);
+    this.entries = entries;
+    int selectionIndex = -1;
+    Location lastSelectedLocation = null;
+
+    if (statsChanged(oldEntries, entries)) {
+      final int selectedRowIndex = getSelectedRow();
+      if (selectedRowIndex != -1) {
+        final Object selectedRow = getTreeModel().getRowValue(selectedRowIndex);
+        if (selectedRow instanceof DefaultMutableTreeNode) {
+          final DefaultMutableTreeNode selectedRowNode = (DefaultMutableTreeNode)selectedRow;
+          final SlidingWindowStatsSummary selectedStats = (SlidingWindowStatsSummary)selectedRowNode.getUserObject();
+          if (selectedStats != null) {
+            lastSelectedLocation = selectedStats.getLocation();
+          }
+        }
+      }
+      // TOOD(jacobr): avoid updating the model when
```

Comment:
Avoid updating the model when...??

Reply:
removed incomplete comment as I fixed it but forgot to update.